### PR TITLE
docs: clarify group fields in config specs

### PR DESCRIPTION
**Pull request by bgilbert**
They specify the group that should be applied to the inode, not the primary group of the inode's owner.

### CODE REVIEW

**Review by jlebon — APPROVED** [review]
Did this cause confusion to someone or just a drive-by thing you noticed?

File: docs/configuration-v3_0.md [hunk 1]
```diff
@@ -72,28 +72,28 @@ The Ignition configuration is a JSON document conforming to the following specif
...
   * **_links_** (list of objects): the list of links to be created. Every file, directory, and link must have a unique `path`.
     * **path** (string): the absolute path to the link
     * **_overwrite_** (boolean): whether to delete preexisting nodes at the path. If overwrite is false and a matching link exists at the path, Ignition will only set the owner and group. Defaults to false.
     * **_user_** (object): specifies the symbolic link's owner.
       * **_id_** (integer): the user ID of the owner.
       * **_name_** (string): the user name of the owner.
-    * **_group_** (object): specifies the group of the owner.
-      * **_id_** (integer): the group ID of the owner.
-      * **_name_** (string): the group name of the owner.
+    * **_group_** (object): specifies the symbolic link's group.
```
Comment (jlebon) [inline docs/configuration-v3_0.md:94]: Hmm, it's only a symbolic link if `hard` is `false`, right?

**Comment by bgilbert** [comment]
No, I just happened to notice this.

File: docs/configuration-v3_0.md (hunk 1 above)
Comment (bgilbert) [reply docs/configuration-v3_0.md:94]: If `hard` is true, we can't set the owner/perms independently from the target, so we don't try.  I'll add another clarification.

**Comment by bgilbert** [comment]
Updated docs, and also added validation warnings if `user` or `group` is used with a hard link.

**Review by prestist — APPROVED** [review]
Good catch, reads a lot better.

**Review by jlebon — APPROVED** [review]

**Pull request merged**